apps/vanilla/src/components/api.ts

```diff
@@ -0,0 +1,55 @@
+import { Pagination } from '@js-camp/core/models/pagination';
+import { PaginationMapper } from '@js-camp/core/mappers/pagination.mapper';
+import { AnimeTypeEnum } from '@js-camp/core/enums/animeType';
+import { AnimeOrderingEnum } from '@js-camp/core/enums/orderingType';
+import Anime from '@js-camp/core/models/anime';
+
+/** A class designed to interact with API. */
+export class API {
+  /** Base URL. */
+  private readonly baseUrl: string;
+
+  /** URL query part. */
+  public urlQuery: string;
+
+  public constructor(baseUrl: string, urlQuery: string) {
+    this.baseUrl = baseUrl;
+    this.urlQuery = urlQuery;
+  }
+
+  /** Send request.
+   * @param url Link.
+   */
+  async getPagination(url: string): Promise<Pagination<Anime>> {
+    try {
+      let res = await fetch(url);
+      let data = await res.json();
```

Comment:
https://wiki.saritasa.rocks/frontend/languages/ts-js/best-practices#promote-immutability
https://wiki.saritasa.rocks/frontend/languages/ts-js/naming#avoid-contractions
```suggestion
      const response = await fetch(url);
      const data = await response.json();
```